itest/lnd_onchain_test.go

```diff
@@ -74,6 +77,100 @@ func testChainKitGetBlockHash(ht *lntest.HarnessTest) {
 	require.Equal(ht, expected, actual)
 }
 
+// testChainKitSendOutputsAnchorReserve checks if the SendOutputs rpc prevents
+// our wallet balance to drop below the required anchor channel reserve amount.
+func testChainKitSendOutputsAnchorReserve(ht *lntest.HarnessTest) {
+	// Start two nodes supporting anchor channels.
+	args := lntest.NodeArgsForCommitType(lnrpc.CommitmentType_ANCHORS)
+
+	// NOTE: we cannot reuse the standby node here as the test requires the
+	// node to start with no UTXOs.
+	alice := ht.NewNode("Alice", args)
+	bob := ht.Bob
+	ht.RestartNodeWithExtraArgs(bob, args)
+
+	// We'll start the test by sending Alice some coins.
+	fundingAmount := btcutil.Amount(100_000)
+	ht.FundCoins(fundingAmount, alice)
+
+	// Before opening the channel we ensure that the nodes are connected.
+	ht.EnsureConnected(alice, bob)
+
+	// We'll get the anchor reserve that is required for a single channel.
+	reserve, _ := alice.RPC.WalletKit.RequiredReserve(
+		context.Background(), &walletrpc.RequiredReserveRequest{
+			AdditionalPublicChannels: 1,
+		},
+	)
+
+	// Alice opens an anchor channel and keeps twice the amount of the
+	// anchor reserve in her wallet.
+	chanAmt := fundingAmount - 2*btcutil.Amount(reserve.RequiredReserve)
+	outpoint := ht.OpenChannel(
+		alice, bob, lntest.OpenChannelParams{
+			Amt:            chanAmt,
+			CommitmentType: lnrpc.CommitmentType_ANCHORS,
+			SatPerVByte:    1,
+		},
+	)
+
+	// Now we obtain a taproot address from bob which Alice will use to send
+	// coins to him via the SendOutputs rpc.
+	address := bob.RPC.NewAddress(&lnrpc.NewAddressRequest{
+		Type: lnrpc.AddressType_TAPROOT_PUBKEY,
+	})
+	decodedAddr := ht.DecodeAddress(address.Address)
+	addrScript := ht.PayToAddrScript(decodedAddr)
+
+	// First she will try to send Bob an amount that would undershoot her
+	// reserve requirement by one satoshi.
+	balance := alice.RPC.WalletBalance()
+	utxo := &wire.TxOut{
+		Value:    balance.TotalBalance - reserve.RequiredReserve + 1,
+		PkScript: addrScript,
+	}
+	req := &walletrpc.SendOutputsRequest{
+		Outputs: []*signrpc.TxOut{{
+			Value:    utxo.Value,
+			PkScript: utxo.PkScript,
+		}},
+		SatPerKw: 2400,
+		MinConfs: 1,
+	}
+
+	// We try to send the reserve violating transaction.
+	_, err := alice.RPC.WalletKit.SendOutputs(context.Background(), req)
+	ht.MineBlocksAndAssertNumTxes(1, 0)
+
+	// We expect SendOutputs to return the respective error.
+	require.ErrorContains(ht, err, "the outputs to be send would leave "+
```

Comment:
Also, we should probably define this as an exported error, then we can string match on it here to shorten the line and also avoid braking the test when the message is updated.